pkg/objstore/azure/azure.go

```diff
@@ -190,15 +190,19 @@ func (b *Bucket) getBlobReader(ctx context.Context, name string, offset, length
 	}
 
 	var size int64
-	if length > 0 {
+	// If a length is specified and it won't go past the end of the file,
+	// then set it as the size
+	if length > 0 && length <= props.ContentLength()-offset {
```

Comment:
This does not look right.

AFAIK there is no known case in immutable system that Thanos would ask for something that does not exists unless TSDB block is malformed (malformed index, index pointing to not existsing chunks). Is that the case?

Even in this case this change would delegate error to Thanos which would see wrong result from Azure anyway.

Reply:
hey @bwplotka, just pinging you for another look.

btw, congrats on the Redhat move! :smile:

Reply:
@bwplotka I can confirm that it works or me too